packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -119,6 +120,11 @@ export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps
             >
               <ApiContext.Provider value={apiRef}>
                 <OptionsContext.Provider value={gridState.options}>
+                  {customComponents.toolbarComponent || (
```

Comment:
you can do the check for hideToolbar here

Reply:
Done